patches/server/0929-Add-Sneaking-API-for-Entities.patch

```diff
@@ -0,0 +1,52 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: dawon <dawon@dawon.eu>
+Date: Wed, 19 Oct 2022 23:13:42 +0200
+Subject: [PATCH] Add Sneaking API for Entities
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
+index 9368ec01e498f913bc5b7b3e77fe87659090d9b5..b80b22fc0982694dee2129193d47025da9010042 100644
+--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
++++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
+@@ -1152,6 +1152,18 @@ public abstract class CraftEntity implements org.bukkit.entity.Entity {
+         return Pose.values()[this.getHandle().getPose().ordinal()];
+     }
+ 
++    // Paper start
++    @Override
++    public void setSneaking(boolean sneak) {
++        this.getHandle().setShiftKeyDown(sneak);
++    }
++
++    @Override
++    public boolean isSneaking() {
++        return this.getHandle().isShiftKeyDown();
++    }
++    // Paper end
++
+     @Override
+     public SpawnCategory getSpawnCategory() {
+         return CraftSpawnCategory.toBukkit(this.getHandle().getType().getCategory());
+diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+index b1136b9c39b16cbb9dfe460f88000f74ccd4f571..e398335056f3d23c1225871bfe557909a8ad9026 100644
+--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
++++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+@@ -1348,12 +1348,16 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
+ 
+     @Override
+     public void setSneaking(boolean sneak) {
+-        this.getHandle().setShiftKeyDown(sneak);
++        // Paper start - move the logic to Entity class
++        super.setSneaking(sneak);
++        // Paper end
+     }
+ 
+     @Override
+     public boolean isSneaking() {
+-        return this.getHandle().isShiftKeyDown();
++        // Paper start - move the logic to Entity class
++        return super.isSneaking();
++        // Paper end
```

Comment:
Is it worth making this change? functionally its no different and so its extra useless diff, but at the same time its smart to have it in one place instead of two.

Reply:
So just put it in Entity and leave the duplicated code in Player?

Reply:
I don't know, which is why it's not a requested change, just a comment. Someone else might think that, but I don't have an opinion either way.

Reply:
Me neither. But your point it will make more diff is valid and knowing how much stuff needs to be updated during update I guess it makes sense to just leave it in there, considering it is oneliner anyway...

Reply:
Yea, if there was a bunch of logic in there that needed to be moved and modified, sure; but, for a singular line that is purely just delegating a call over elsewhere, it's just pointless diff